Doc: Add Known issue for 8.18.7: Agent stuck on failed upgrade

karenzone opened the pull request:
Closes: https://github.com/elastic/docs-content/issues/2978

Adds known issue to docs: "Failed upgrades leave agent stuck until restart"

karenzone committed:
Doc: Add KI 8.18.7: Agent stuck on failed upgrade

karenzone commented:
<img width="699" height="1027" alt="Screenshot 2025-09-17 at 8 08 05 PM" src="https://github.com/user-attachments/assets/10c8cf1c-8d38-47c4-8db6-98cd2dee8468" />

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+[%collapsible]
+====
+
+This known issue applies to {agent} 8.18.7 and 9.0.7. {agent} versions 8.19.x and 9.1.x are not affected.
```
Are versions correct?

pkoutsovasilis replied:
yep 👍

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+This known issue applies to {agent} 8.18.7 and 9.0.7. {agent} versions 8.19.x and 9.1.x are not affected.
+
+On September 17, 2025, a known issue was discovered that can cause {agent} upgrades to get stuck if an upgrade attempt fails early. This happens because the coordinator's overrideState remains set, leaving the agent in a state that appears to be upgrading.
```
Is discovery date correct?

pkoutsovasilis replied:
yep 👍

karenzone commented:
As @pkoutsovasilis @cmacknz, please add anybody else you'd like to review.

cmacknz commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+This known issue applies to {agent} 8.18.7 and 9.0.7. {agent} versions 8.19.x and 9.1.x are not affected.
+
+On September 17, 2025, a known issue was discovered that can cause {agent} upgrades to get stuck if an upgrade attempt fails early. This happens because the coordinator's overrideState remains set, leaving the agent in a state that appears to be upgrading.
```
```suggestion
On September 17, 2025, a known issue was discovered that can cause {agent} upgrades to get stuck if an upgrade attempt fails under specific conditions. This happens because the coordinator's overrideState remains set, leaving the agent in a state that appears to be upgrading.
```

I think it's important to emphasize this problem does not happen for any upgrade failed, like the most common cases where the download fails.

cmacknz commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+- The agent is not upgradeable
+- Capabilities check denies the upgrade
+- Most commonly: When {agent} is tamper-protected and Endpoint returns an error during action proxying, for example, because the upgrade action signature is invalid, missing, or fails verification. This causes the coordinator's override state to be stuck.
```
```suggestion
- Most commonly: When {agent} is tamper-protected and Endpoint fails to validate that the upgrade action was correctly signed by Kibana to allow the upgrade, for example, because the signature is missing, invalid, or the connection between {agent} and endpoint was interrupted. This causes the agent coordinator's override state to become stuck until the agent is restarted.
```

Trying to reword this a bit so users can understand the specific circumstances more clearly.

karenzone replied:
@cmacknz, what do you this about this tweak to your suggestion?  I'm trying to capture the extra context that you provided, while making the content easier to read and parse. 

_"When {agent} is tamper-protected, Endpoint must validate that the upgrade action was correctly signed by Kibana to allow the upgrade. If the signature is missing, invalid, or the connection between {agent} and Endpoint was interrupted, the validation fails. This causes the agent coordinator's override state to become stuck until the agent is restarted."_

cmacknz replied:
Looks great, thanks!

karenzone committed:
Apply suggestions from code review

Co-authored-by: Craig MacKenzie <craig.mackenzie@elastic.co>

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+This known issue applies to {agent} 8.18.7 and 9.0.7. {agent} versions 8.19.x and 9.1.x are not affected.
+
+On September 17, 2025, a known issue was discovered that can cause {agent} upgrades to get stuck if an upgrade attempt fails under specific conditions. This happens because the coordinator's overrideState remains set, leaving the agent in a state that appears to be upgrading.
```
```suggestion
On September 17, 2025, a known issue was discovered that can cause {agent} upgrades to get stuck if an upgrade attempt fails under specific conditions. This happens because the coordinator's `overrideState` remains set, leaving the agent in a state that appears to be upgrading.
```

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+- The agent is not upgradeable
+- Capabilities check denies the upgrade
+- Most commonly: When {agent} is tamper-protected and Endpoint fails to validate that the upgrade action was correctly signed by Kibana to allow the upgrade, for example, because the signature is missing, invalid, or the connection between {agent} and endpoint was interrupted. This causes the agent coordinator's override state to become stuck until the agent is restarted.
```
```suggestion
This issue is triggered if the upgrade fails during one of the early checks inside `Coordinator.Upgrade`, for example:

- The agent is not upgradeable
- Capabilities check denies the upgrade
- When {agent} is tamper-protected, Endpoint must validate that the upgrade action was correctly signed by Kibana to allow the upgrade. If the signature is missing, invalid, or the connection between {agent} and Endpoint was interrupted, the validation fails. This causes the agent coordinator's override state to become stuck until the agent is restarted.
```

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+**Workaround**
+
+Restart the {agent} to clear the coordinator's overrideState and allow new upgrade attempts to proceed.
```
```suggestion
- Elastic Agent status shows an override state indicating upgrade 

**Workaround**

Restart the {agent} to clear the coordinator's `overrideState` and allow new upgrade attempts to proceed.
```

karenzone commented on docs/en/ingest-management/release-notes/release-notes-8.18.asciidoc:
```diff
@@ -33,6 +33,45 @@ Also see:
+
+This issue was fixed in link:https://github.com/elastic/elastic-agent/pull/9992[#9992], which ensures that the coordinator clears its override state whenever an early failure occurs.
+
+This fix will be included in versions 9.1.4, 8.19.4, 9.0.8, and 8.18.8.
```
```suggestion
The fix is included in versions 9.1.4 and 8.19.4, and planned for versions 9.0.8 and 8.18.8.
```

karenzone reviewed (commented):
Porting over review comments from 9.0.7 known issue pr (elastic-agent#10042)

karenzone committed:
Porting over review changes from 9.0.7 known issue PR

karenzone commented:
I added review changes from the [9.0.7 known issue PR](https://github.com/elastic/elastic-agent/pull/10042) here, and added review changes from this PR to the 9.0.7 PR.  ✅
